raiden/channel.py

```diff
@@ -428,16 +430,19 @@ def set_closed(self, block_number):
             callback(block_number)
 
     def set_settled(self, block_number):
-        if self._settled_block != 0:
-            raise RuntimeError('channel is already settled')
+        # ensure callbacks are only called once
+        if self._settled_block != 0 and self._settled_block != block_number:
+            raise RuntimeError('channel is already settled on different block %s %s'
+                               % (self._settled_block, block_number))
+        else:
+            self._settled_block = block_number
 
-        self._settled_block = block_number
-
-        for callback in self.callbacks_settled:
-            callback(block_number)
+            for callback in self.callbacks_settled:
+                callback(block_number)
 
     def query_settled(self):
-        return self.netting_channel.settled()
+        # FIXME: the if None: return 0 constraint should be ensured on the proxy side
```

Comment:
see https://travis-ci.org/raiden-network/raiden/jobs/201871014#L477 of the demo PR #398